sdk/dotnet/Network/NetworkSecurityRule.cs

```diff
@@ -247,11 +247,17 @@ public InputList<string> DestinationAddressPrefixes
             set => _destinationAddressPrefixes = value;
         }
 
+        [Input("destinationApplicationSecurityGroupIds")]
+        private InputList<string>? _destinationApplicationSecurityGroupIds;
+
         /// <summary>
         /// A List of destination Application Security Group ID's
         /// </summary>
-        [Input("destinationApplicationSecurityGroupIds")]
-        public Input<string>? DestinationApplicationSecurityGroupIds { get; set; }
+        public InputList<string> DestinationApplicationSecurityGroupIds
```

Comment:
Some breaking changes here and below. Let's triage.

Reply:
@mikhailshilkov this is an actual change in the provider

https://github.com/terraform-providers/terraform-provider-azurerm/pull/9884/files

Reply:
I guess we just changelog this and go ahead?